File: stream-chat-android-docs/src/main/java/io/getstream/chat/docs/java/ui/guides/ProvidingCustomReactions.java
package io.getstream.chat.docs.java.ui.guides;

import android.content.Context;

import androidx.core.content.ContextCompat;

import java.util.HashMap;
import java.util.Map;

import io.getstream.chat.android.ui.ChatUI;
import io.getstream.chat.android.ui.helper.SupportedReactions;
import io.getstream.chat.docs.R;

/**
 * [Providing Custom Reactions](https://getstream.io/chat/docs/sdk/android/ui/guides/providing-custom-reactions/)
 */
public class ProvidingCustomReactions {

    public void providingCustomReactions(Context context) {
        Map<String, SupportedReactions.ReactionDrawable> reactions = new HashMap<>();
        reactions.put(
                "thumbs_up", new SupportedReactions.ReactionDrawable(
                        ContextCompat.getDrawable(context, R.drawable.ic_thumb_up),
                        ContextCompat.getDrawable(context, R.drawable.ic_thumb_up_selected)
                )
        );

        reactions.put(
                "thumbs_down", new SupportedReactions.ReactionDrawable(
                        ContextCompat.getDrawable(context, R.drawable.ic_thumb_down),
                        ContextCompat.getDrawable(context, R.drawable.ic_thumb_down_selected)
                )
        );

        reactions.put(
                "mood_good", new SupportedReactions.ReactionDrawable(
                        ContextCompat.getDrawable(context, R.drawable.ic_mood_good),
                        ContextCompat.getDrawable(context, R.drawable.ic_mood_good_selected)
                )
        );

        reactions.put(
                "mood_bad", new SupportedReactions.ReactionDrawable(
                        ContextCompat.getDrawable(context, R.drawable.ic_mood_bad),
                        ContextCompat.getDrawable(context, R.drawable.ic_mood_bad_selected)
                )
        );

        ChatUI.setSupportedReactions(new SupportedReactions(context, reactions));
    }
}

File: stream-chat-android-docs/src/main/kotlin/io/getstream/chat/docs/kotlin/ui/guides/ProvidingCustomReactions.kt
// ktlint-disable filename

package io.getstream.chat.docs.kotlin.ui.guides

import android.content.Context
import androidx.core.content.ContextCompat
import io.getstream.chat.android.ui.ChatUI
import io.getstream.chat.android.ui.helper.SupportedReactions
import io.getstream.chat.docs.R

/**
 * [Providing Custom Reactions](https://getstream.io/chat/docs/sdk/android/ui/guides/providing-custom-reactions/)
 */
private object ProvidingCustomReactionsSnippet {

    fun providingCustomReactions(context: Context) {
        val reactions = mapOf(
            "thumbs_up" to SupportedReactions.ReactionDrawable(
                inactiveDrawable = ContextCompat.getDrawable(context, R.drawable.ic_thumb_up)!!,
                activeDrawable = ContextCompat.getDrawable(context, R.drawable.ic_thumb_up_selected)!!
            ),
            "thumbs_down" to SupportedReactions.ReactionDrawable(
                inactiveDrawable = ContextCompat.getDrawable(context, R.drawable.ic_thumb_down)!!,
                activeDrawable = ContextCompat.getDrawable(context, R.drawable.ic_thumb_down_selected)!!
            ),
            "mood_good" to SupportedReactions.ReactionDrawable(
                inactiveDrawable = ContextCompat.getDrawable(context, R.drawable.ic_mood_good)!!,
                activeDrawable = ContextCompat.getDrawable(context, R.drawable.ic_mood_good_selected)!!
            ),
            "mood_bad" to SupportedReactions.ReactionDrawable(
                inactiveDrawable = ContextCompat.getDrawable(context, R.drawable.ic_mood_bad)!!,
                activeDrawable = ContextCompat.getDrawable(context, R.drawable.ic_mood_bad_selected)!!
            ),
        )

        ChatUI.supportedReactions = SupportedReactions(context, reactions)
    }
}